add: *not really* material design 3 (nmd3) web UI

RainyXeon opened the pull request:
## Changes

This commit will add webui support for ReZygisk

## Why 

With webui, user will be easier to manage ReZygisk

## Checkmarks

- [x] The modified functions have been tested.
- [x] Used the same indentation as the rest of the project.
- [x] Updated documentation (changelog).

## Additional information

If you have any additional information, write it here

RainyXeon committed:
add: base for webui support in rezygisk

This commit will add the base for webui in rezygisk

RainyXeon committed:
move: webui -> webroot; improve: design for webui

This commit will move webui dir to webroot dir and redesign the webui base on KernelSU UI

ThePedroo committed:
fix: building process; add: web UI JavaScript code

This commit both fixes the building process to include the webroot folder and adds the web UI JS code to allow it to fetch the values.

RainyXeon committed:
add: mark icon for rezygisk state; improve: global fonts

This commit will add new icon for ReZygisk is mark. This icon will replect the state of ReZygisk. This commit also improve font by adding new local font named ProductSans for material you feeling

RainyXeon committed:
improve: description font

This commit will improve description font by adding more margin to make it more clear

RainyXeon committed:
improve: ReZygisk webui state

This commit will improve ReZygisk webui state by adding new red color and hide settings/modules card if it's error or not exist

RainyXeon committed:
remove: useless div card

This commit will remove useless div card in webui

ThePedroo committed:
improve: web UI JS code

This commit improves the web UI Javascript code.

ThePedroo committed:
add: webroot files to get sign

This commit simply adds the webroot files to be also verified the autencity.

RainyXeon committed:
add: light theme base

This commit will add the base of light theme support in ReZygisk WebUI

RainyXeon committed:
add: light theme with button based

RainyXeon committed:
fix: light theme on dim card

RainyXeon committed:
remove: unused code from index.html

RainyXeon committed:
improve: dim card color in light theme

RainyXeon committed:
fix: svg sometimes not showing on webui

RainyXeon committed:
add: new module list interface

RainyXeon committed:
add: pre-fix for module list

RainyXeon committed:
add: loading screen, expandable list and card header icons

This commit will add new loading screen, expandable list and card header icons for improving UX/UI

RainyXeon committed:
fix: module list field now showing on production

RainyXeon committed:
improve: loading screen based in setting up time

RainyXeon committed:
add: start/stop/restart daemon button

RainyXeon committed:
fix: button in light theme

ThePedroo committed:
fix: missing module list

This commit fixes the missing module list due to mismatch in ID names.

ThePedroo committed:
improve: JS code by matching with the standard

This commit improves the JS code by making compliant to the standard of PerformanC.

RainyXeon committed:
add: expand-able menu on settings card; fix: modules list not showing

RainyXeon committed:
add: base code for modal/lang support [W.I.P]

RainyXeon committed:
improve: warn color of bright card

RainyXeon committed:
add: lang list for lang modal

ThePedroo committed:
add: `pt_BR` translation; improve: code syntax

This commit both adds brazilian portuguese translation and improves the code syntax to match the standard.

ThePedroo committed:
add: translation for errors; fix: `pt_BR` not exported

This commit adds translation for errors for both en_US and pt_BR. Also fixes the issue where brazilian portuguese translation wouldn't be exported.

ThePedroo committed:
improve: Web UI by ignoring hw unsupported 64/32-bit

This commit improves Web UI Javascript code by allowing it to be aware of unsupported 64 or 32 bit, thus removing zygoteXX and daemonXX info.

RainyXeon committed:
add: base functional for lang system; fix: weird issue related to js

ExtremeXT committed:
add: `ro_RO` language to web UI (#17)

This commit adds Romanian to the web UI language list.

Emulond committed:
add: `ru_RU` language to web UI (#18)

This commit adds Russian to the web UI language list.

Signed-off-by: Emulond Argent <108662981+Emulond@users.noreply.github.com>

ThePedroo committed:
fix: infinite loading; add: translate on set

This commit fixes the infinite loading area and adds the feature that it will already change all phrases on set, without the need of restart.

RainyXeon committed:
fix: translation for webui for 95% translated

RainyXeon committed:
add: bottom slide animation for modal

This commit will add animation for modal that commit from the bottom (quite goofy) and click animation for theme-switch language-switch and close-modal icon

RainyXeon committed:
improve: modal animation on webui by slow it down

RainyXeon committed:
improve: language system based on json file

This commit will introduce new language system based on json file for easier to translate using crowdin in the future, also fix webui cannot load

RainyXeon committed:
fix: wrong locate of translate

RainyXeon committed:
fix: corrupted zip file in webui (not confirmed)

RainyXeon committed:
fix: corrupted zip file in webui

RainyXeon committed:
fix: cannot load webui

RainyXeon committed:
fix: language list on modal not synchronized

Fyphen1223 committed:
add: japanese translation

RainyXeon committed:
fix: webui infinity loading

RainyXeon committed:
fix: rezygisk appears as demo ui

RainyXeon committed:
add: error history modal for easier to view log and copy

RainyXeon committed:
fix: corrupted zip file in webui

meltartica committed:
add: zn_CN and zn_TW web UI languages (#21)

This commit adds Chinese translations to the web UI.

Signed-off-by: WinCS <94188592+Meltartica@users.noreply.github.com>

ThePedroo committed:
add: credit for Chinese translations

This commit adds the credits for Chinese translations in "TRANSLATOR.md" file.

RainyXeon committed:
add: builtin ProductSans font and option to toggle

meltartica committed:
fix: `zh_CN` and `zh_TW` translations contributor link; improve: complexity in customize.sh

This commit fixes the contributor link for Chinese translations and improves the complexity in customise.sh by reducing it.

RainyXeon committed:
add: track webui error and log directly into error history (may not working)

RainyXeon committed:
remove: webuiError file to avoid inf loading

RainyXeon committed:
add: action card sepreate with settings

RainyXeon committed:
remove: webui test code in js/main.js

RainyXeon committed:
add: new webui framework (using navbar and refactor lang file)

RainyXeon committed:
fix: flash file corrupted

RainyXeon committed:
fix: flash file corrupted (silly error)

RainyXeon committed:
fix: infinity loading

RainyXeon committed:
improve: module element style

RainyXeon committed:
fix: empty screen bug

RainyXeon committed:
fix: empty screen bug (silly error)

RainyXeon committed:
fix: cannot enable light theme

Co-authored-by: CaptainThrowback <captainthrowback@hotmail.com>

RainyXeon committed:
fix: overflow-y cannot scroll

ThePedroo committed:
add: LICENSE for `webroot`

This commit adds proper licensing for webroot folder.

Signed-off-by: Pedro.js <pedroolimpioguerra@gmail.com>

ThePedroo committed:
fix: translations by reverting some to English

This commit reverts the translations done by Google Translator to English, waiting for natives to translate them.

ThePedroo committed:
update: Brazilian portuguese translations

This commit updates the translations for Brazilian portuguese.

ThePedroo committed:
fix: JS code indentation/syntax

This commit fixes the syntax and indentation of the JavaScript code to meet the organization standard.

ThePedroo committed:
add: errors to error log history

This commit allows the errors to be saved on error log history.

ThePedroo committed:
fix: JS code indentation/syntax

This commit fixes the syntax and indentation of the JavaScript code to meet the organization standard.

ThePedroo committed:
add: functional daemon action buttons

This commit adds the function that allows functional daemon action buttons.

ThePedroo committed:
update: WebUI English translation of `Arch` to `Architecture`

This commit changes the word that references the architecture from the brief "Arch" to the full "Architecture" word.

RainyXeon committed:
remove: translation for error log

RainyXeon committed:
add: realtime error log tracking modal

RainyXeon committed:
add: clear all log button

RainyXeon committed:
add: window onerror event check

RainyXeon committed:
add: window.onerror event

RainyXeon committed:
remove: unused zip file

RainyXeon committed:
improve: WebUI by sync card between tab

RainyXeon committed:
improve: action page design; add: daemon status

RainyXeon committed:
fix: reverse zygote status

RainyXeon committed:
add: animation for tab and navbar

RainyXeon committed:
add: credits and license card in settings page

RainyXeon committed:
remove: unused css style

RainyXeon committed:
fix: link open on webui not browser (not confirmed)

RainyXeon committed:
fix: open link on webui instead of browser

RainyXeon committed:
add: fatal error screen

RainyXeon committed:
improve: fatal error screen design

ThePedroo committed:
fix: error upon opening web UI without any zygisk modules

This commit fixes the toast error that appears when opening ReZygisk's web UI.

RainyXeon committed:
add: small page internal API and system based theme

yandev code style on webroot/js/smallPage/theme.js, need to be cleaned

RainyXeon committed:
remove: flag from pt_BR translation

RainyXeon committed:
fix: webui error when no zygigk module installed

RainyXeon committed:
improve: no module avaliable div

RainyXeon committed:
remove: unused string

ThePedroo committed:
update: Brazilian portuguese translations

This commit updates the translations of Brazilian portuguese of WebUI.

Signed-off-by: Pedro.js <pedroolimpioguerra@gmail.com>

RainyXeon committed:
fix: light icon for navbar in system based theme (not confirmed)

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

RainyXeon committed:
remove: useless fix for navbar

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

RainyXeon committed:
fix: light theme navbar

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

ThePedroo committed:
remove: duplicated logs

This commit removes duplication of logs in ptrace binaries.

ThePedroo committed:
fix: error upon loading Web UI

This commit fixes issues that caused the Web UI to not load.

ThePedroo committed:
fix: untranslated string in Brazilian portuguese

This commit fixes the missing translations for Brazilian portuguese.

Signed-off-by: Pedro.js <pedroolimpioguerra@gmail.com>

miuirussia committed:
fix #36: WebUI not working on 64 bit only devices

ThePedroo committed:
improve: Web UI loading speed

This commit improves the speed of loading of the Web UI.

ThePedroo committed:
fix: `pt_BR` translation grammar

This commit fixes grammar mistakes in brazilian portuguese translation.

SheepChef committed:
improve: chinese translation. (#42)

* Full Chinese Translation.

Translate all currently existing English strings to Chinese.

Signed-off-by: SheepChef <50871867+SheepChef@users.noreply.github.com>

* Update language.js

---------

Signed-off-by: SheepChef <50871867+SheepChef@users.noreply.github.com>

SheepChef committed:
fix: `lines` variable not declared; improve: style of web UI (#43)

This commit fixes an issue where "lines" variable would not be declared and improves the style of the web UI.

Signed-off-by: SheepChef <50871867+SheepChef@users.noreply.github.com>
Co-authored-by: Pedro.js <pedroolimpioguerra@gmail.com>

RainyXeon committed:
fix: translation credits synchronization

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

RainyXeon committed:
add: credits for SheepChef on TRANSLATOR.md

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

AJleKcAHgP68 committed:
improve: russian translation on ru_RU.json (#44)

Add RU translate

Signed-off-by: AJleKcAHgP68 <78802768+AJleKcAHgP68@users.noreply.github.com>

RainyXeon committed:
add: AJleKcAHgP68's russian translation credits

This commit will add AJleKcAHgP68's russian translation credits to TRANSLATION.md and webroot/js/language.js

Co-authored-by: AJleKcAHgP68 <78802768+AJleKcAHgP68@users.noreply.github.com>

RainyXeon committed:
add: initial support for tablet/desktop mode

This commit will add initial support for tablet/desktop mode by trick the navbar a bit

RainyXeon committed:
improve: tablet navbar by decrease scale

RainyXeon committed:
add: outfit font to replace product sans

RainyXeon committed:
fix: build failed for webui

ThePedroo committed:
fix: `zygisk-ptraceXX` response parsing

This commit fixes the parsing of "ptrace" command.

RainyXeon committed:
add: amoled and monochrome theme

This commit will add 2 new themes is monochrome and amoled

RainyXeon committed:
remove: test code

Blazzycrafter committed:
add: `de_DE` translation credits (#52)

This commit adds the credit for Blazzycrafter, the contributor whom added German translations for ReZygisk WebUI.

Signed-off-by: Blazzycrafter <39300111+Blazzycrafter@users.noreply.github.com>

ygorigor committed:
update: `ro_RO` translations (#57)

This commit updates Romanian translations to translate untranslated phrases and fix some grammatically incorrect phrases.

Signed-off-by: Igor <sorocean.igor@gmail.com>

Blazzycrafter committed:
add: `de_DE` translations (#62)

This commit adds German translations for ReZygisk WebUI.

Signed-off-by: Blazzycrafter <39300111+Blazzycrafter@users.noreply.github.com>

ThePedroo committed:
improve: translations detection

This commit allows ReZygisk WebUI to dynamically fetch translations.

RainyXeon committed:
add: new cache state storing and handling

ThePedroo committed:
update: ReZygisk's Web UI license

This commit updates the Web UI of ReZygisk to a license with copyleft, making it harder to be used in proprietary projects.

ThePedroo committed:
fix: Web UI not loading

This commit fixes the issue where the Web UI doesn't load by properly implementing the saving of the status of ReZygisk so that it can be displayed easily, which wasn't properly implemented before.

RainyXeon committed:
add: new ui for error history page

remove: modal support (initial)

RainyXeon committed:
fix: error history header

ThePedroo committed:
fix: re-declaring constant variable

This commit fixes the issue where two variables with the same name would overlap, causing a re-declaration of a constant (imutable) variable

RainyXeon committed:
remove: button icon in light theme

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

RainyXeon committed:
remove: light icon data for errorh page

Signed-off-by: unexpected unresolved <minh15052008@gmail.com>

RainyXeon committed:
fix: go back icon not change the color

RainyXeon committed:
add: some small padding for header

RainyXeon committed:
fix: translation not completely for webui

SheepChef committed:
update: `zh_CN` translations (#78)

This commit updates the Simplified Chinese translations for the WebUI.

Signed-off-by: SheepChef <50871867+SheepChef@users.noreply.github.com>

ThePedroo committed:
fix: 64-bit/32-bit-only device detection

This commit fixes the detection made by ReZygisk WebUI that makes it be aware of whether the device's CPU is capable of 64-bit or 32-bit only, allowing it to properly mark as Fully functioning or just partially.

ThePedroo committed:
fix: monitor information

This commit fixes monitor information, which now is properly working and named, and also has proper status words.

RainyXeon committed:
add: new translation for vi_VN

Signed-off-by: unexpected unresolved <xeondev@xeondex.onmicrosoft.com>

SheepChef committed:
update: translations for `zh_CN` (#79)

This commit updates the translations for Simplified Chinese so that it is fully translated.

Signed-off-by: SheepChef <50871867+SheepChef@users.noreply.github.com>

RainyXeon committed:
fix: language cannot translated fully

RainyXeon committed:
fix: wrong file name (build)

Signed-off-by: unexpected unresolved <xeondev@xeondex.onmicrosoft.com>

RainyXeon committed:
fix: wrong file name (installer)

Signed-off-by: unexpected unresolved <xeondev@xeondex.onmicrosoft.com>

bpanca05 committed:
add: indonesian language for WebUI (#83)

This commit will add support for indonesian language in webui, following commits:

* Add Indonesian language for WebUI

* Sort TRANSLATOR.md contributor alphabetically

witchfuneral committed:
add: Turkish language for WebUI (#85)

Co-authored-by: witch <witch@dyingwillow.dev>

miuirussia committed:
fix: `ru_RU` language name translation (#86)

This commit fixes russian language name translation, which previously said, in russian, "English (USA)".

Signed-off-by: Kirill Kuznetsov <kdevlab@yandex.ru>

GuitarHeroStyles committed:
update: `id_ID` translations (#107)

This commit updates the translations for Indonesia language with proper grammar.

Signed-off-by: Lucky Kiddos <95188840+GuitarHeroStyles@users.noreply.github.com>

miuirussia commented:
Rebase with main?

ThePedroo commented:
?

miuirussia commented:
I use build from this branch, please rebase it with changes of main branch. So many changes in main

ThePedroo commented:
I'm confused of what you mean as there are no conflicts, so supposedly it can be merged flawlessly.

Kittyskj committed:
add: ukrainian translations to WebUI (#100)

This commit adds Ukrainian translations for the WebUI available languages.

Signed-off-by: Kitty <73357005+Kittyskj@users.noreply.github.com>
Co-authored-by: Pedro.js <pedroolimpioguerra@gmail.com>

GhostFRR committed:
add: `@GhostFRR`  to `TRANSLATOR.md`  (#134)

This commit adds "@GhostFRR" to the "TRANSLATOR.MD" credits for translating the WebUI strings to French.

Signed-off-by: GhostFRR <ghost.game.fr@gmail.com>

ThePedroo committed:
fix: merge conflicts

This commit fixes merge conflicts while merging to main.

ThePedroo committed:
improve: use module name, not module id; add: show Android/Kernel version in WebUI

This commit improves the WebUI module listing by listing the module name rather than the module id, and also adds 2 fields in the WebUI, so that it shows the Android version and Kernel release string.

ThePedroo committed:
remove: `webroot` data from `machikado` creation

This commit removes "webroot" files from being used for "machikado" file creation, as it doesn't affect core functionality. For safety a new approach will be created later.

ThePedroo committed:
improve: reduce size of section headers

This commit reduce the size of section headers to make it more appropriate.

RainyXeon committed:
improve: some small space on section header

RainyXeon committed:
improve: spacing on Rezygisk section header

Signed-off-by: Alice w/🌧️ <rainyxeon@gmail.com>

RainyXeon committed:
fix: misconfiguration on section header

Signed-off-by: Alice w/🌧️ <rainyxeon@gmail.com>

RainyXeon committed:
add: debug log for chrome dev tools (don't remove)

RainyXeon committed:
fix: debug log

ThePedroo committed:
fix: WebUI ReZygisk info retrieval

This commit fixes how the WebUI retrieves ReZygisk information, such as if it is working, which modules are loaded, etc.

ZG089 committed:
add: Arabic Translation (#150)

* feat: Added Arabic Translation

Signed-off-by: ＺＧＸ０８９ッ <159061718+ZG089@users.noreply.github.com>

* Update TRANSLATOR.md

Signed-off-by: ＺＧＸ０８９ッ <159061718+ZG089@users.noreply.github.com>

---------

Signed-off-by: ＺＧＸ０８９ッ <159061718+ZG089@users.noreply.github.com>

RainyXeon committed:
add: experiment RTL layout feature

RainyXeon committed:
add: temporary fix for translation as some code in home translate broken

RainyXeon committed:
add: ar_EG to extractor

Signed-off-by: Alice w/🌧️ <rainyxeon@gmail.com>

ThePedroo committed:
update: translations; add: new translations; remove: incomplete translations

This commit updates and adds new translations based on contributions on Crowdin. If you are the contributor of any of the added ones, or participated in the updates of any, please, submit a PR adding your proper credit as the "README.md" suggests. Also removes translations that are incomplete at the moment of this commit. The credits for those will not be removed.

ZG089 committed:
fix: some translation in ar_EG.json (#153)

Signed-off-by: ＺＧＸ０８９ッ <159061718+ZG089@users.noreply.github.com>

ThePedroo committed:
fix: WebUI loaded modules list parsing

This commit fixes the code that parses the loaded modules from ReZygisk "module.prop", which previously would only load the first module as it used "info[1]".

ThePedroo committed:
improve: `webroot` extraction

This commit improves the "webroot" folder extraction in "customize.sh" by simplifying into a single unzip command.

thasave14 committed:
add: `it_IT` translations for the WebUI (#154)

This commit adds Italian translations for ReZygisk WebUI.

Signed-off-by: thasave14 <93542339+thasave14@users.noreply.github.com>

GhostFRR committed:
add: `fr_FR.json` translations to WebUI (#156)

This commit adds French translations to ReZygisk WebUI.

Signed-off-by: GhostFRR <ghost.game.fr@gmail.com>

ThePedroo committed:
add: `es_AR` translations to the WebUI; fix: `fr_FR` file extension

This commit adds the Argentina Spanish translations to ReZygisk WebUI and fixes French translations file not having ".json".

Co-authored-by: Flopster101 <nahuelgomez329@gmail.com>

Lxchoooo committed:
add: `es_ES` and `es_MX` translations to the WebUI (#157)

This commit adds both Spanish and Mexico Spanish translations to the ReZygisk WebUI.

GuitarHeroStyles committed:
update: WebUI `id_ID` translations (#160)

This commit updates WebUI Indonesian translation to improve clarity.

Signed-off-by: Lucky Kiddos <95188840+GuitarHeroStyles@users.noreply.github.com>

ThePedroo committed:
fix: WebUI translations file names and line ending

This commit fixes the file names of some translations of WebUI strings and the missing new space in their end.

ThePedroo committed:
fix: improper handling of `unknown`/`exited` zygote status in WebUI

This commit fixes the issue where the daemon info would not be found, causing a crash in WebUI due to accessing properties of "undefined" object.

ThePedroo committed:
Merge branch 'main' into add/new-webui

Signed-off-by: Pedro.js <pedroolimpioguerra@gmail.com>

ThePedroo reviewed (approved):
LGTM

ThePedroo committed:
fix: extracting sha256 sum of `webroot` files

This commit fixes the issue where "webroot" files would have its sha256 hash extracted to the installation directory by using unzip "-x" flag to exclude ".sha256" files with that.

ThePedroo committed:
improve: immediatly set monitor status after set in WebUI

This commit improves the "Actions" tab by immediatly changing visually the monitor status in the WebUI once modified.
